compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -3436,11 +3436,28 @@ public BLangNode transform(ArrayTypeDescriptorNode arrayTypeDescriptorNode) {
             } else {
                 Node keyExpr = arrayTypeDescriptorNode.arrayLength().get();
                 if (keyExpr.kind() == SyntaxKind.NUMERIC_LITERAL) {
+                    int length;
                     Token literalToken = ((BasicLiteralNode) keyExpr).literalToken();
                     if (literalToken.kind() == SyntaxKind.DECIMAL_INTEGER_LITERAL_TOKEN) {
-                        sizes.add(new BLangLiteral(Integer.parseInt(literalToken.text()), symTable.intType));
+                        long lengthCheck = Long.parseLong(literalToken.text());
+                        if (lengthCheck > Integer.MAX_VALUE - 10) { // due to the JVM limitations
```

Comment:
Let's extract `Integer.MAX_VALUE - 10` into a `final static long` field, and add a comment explaining why you choose  `- 10`

Reply:
okay. I'll change it.